python/dgl/graphbolt/itemset.py

```diff
@@ -255,6 +267,44 @@ def __iter__(self) -> Iterator:
     def __len__(self) -> int:
         return sum(len(itemset) for itemset in self._itemsets.values())
 
+    def __getitem__(self, idx: Union[int, slice, Iterable]) -> Dict[str, Tuple]:
+        try:
+            len(self)
+        except TypeError:
+            raise TypeError(
+                f"{type(self).__name__} instance doesn't support indexing."
+            )
+        offsets = [len(itemset) for itemset in self._itemsets.values()]
+        total_num = sum(offsets)
```

Comment:
I thought about it. pros of moving to init is to avoid `sum` in runtime. cons is it needs to check  if `len` supported and init if supported. multiple lines of code will be added into `__init__`. I don't think it's elegant. And the reason why I leave it here is the overhead should be trivial: instantiate offset and sum it.

Reply:
moved to `__init__`